app/models/policy.rb

```diff
@@ -44,40 +54,38 @@ def possible_nations
   end
 
   def applicable_nations
-    applicable_nations = possible_nations.select { |n|
-      self.send(n) == true
-    }
+    possible_nations.select { |n| self.send(n) == true }
   end
 
   def inapplicable_nations
     possible_nations - applicable_nations
   end
 
-  def organisations
-    organisation_content_ids.map { |content_id| find_organisation(content_id) }.compact
-  end
-
-  def people
-    people_content_ids.map { |content_id| find_person(content_id) }.compact
+  def organisations(fetcher = ContentItemFetcher.new)
+    @fetched_organisations ||= organisation_content_ids.map { |content_id| fetcher.find_organisation(content_id) }.compact
   end
 
-  def working_groups
-    working_group_content_ids.map { |content_id| find_working_group(content_id) }.compact
+  def people(fetcher = ContentItemFetcher.new)
+    @fetched_people ||= people_content_ids.map { |content_id| fetcher.find_person(content_id) }.compact
   end
 
-private
-  def find_person(content_id)
-    ContentItemFetcher.people.find { |person| person["content_id"] == content_id }
+  def working_groups(fetcher = ContentItemFetcher.new)
+    @fetched_working_groups ||= working_group_content_ids.map { |content_id| fetcher.find_working_group(content_id) }.compact
   end
 
-  def find_organisation(content_id)
-    ContentItemFetcher.organisations.find { |organisation| organisation["content_id"] == content_id }
+  # Fetch links from the publisher-api
+  def fetch_links!
+    if content_id.nil?
```

Comment:
You can single line this to `return if content_id.blank?`. 

`blank?` is better than `nil?` in this case as you want to detect the case where this is empty string as well.